cmake/SetupCodeChecks.cmake

```diff
@@ -412,15 +453,96 @@ macro(blt_add_astyle_target)
 
     # Check/Set required parameters
     if(NOT DEFINED arg_NAME)
-        message(FATAL_ERROR "blt_add_astyle_target requires a NAME parameter")
+        message(FATAL_ERROR "blt_add_clangformat_target requires a NAME parameter")
     endif()
 
     if(NOT DEFINED arg_CFG_FILE)
-        message(FATAL_ERROR "blt_add_astyle_target requires a CFG_FILE parameter")
+        message(FATAL_ERROR "blt_add_clangformat_target requires a CFG_FILE parameter")
     endif()
 
     if(NOT DEFINED arg_SRC_FILES)
-        message(FATAL_ERROR "blt_add_astyle_target requires a SRC_FILES parameter")
+        message(FATAL_ERROR "blt_add_clangformat_target requires a SRC_FILES parameter")
+    endif()
+
+    if(NOT DEFINED arg_MODIFY_FILES)
+        set(arg_MODIFY_FILES FALSE)
+    endif()
+
+    if(DEFINED arg_WORKING_DIRECTORY)
+        set(_wd ${arg_WORKING_DIRECTORY})
+    else()
+        set(_wd ${CMAKE_CURRENT_SOURCE_DIR})
+    endif()
+
+    set(_generate_target TRUE)
+
+    # Copy config file to given working directory since ClangFormat doesn't support pointing to one
+    configure_file(${arg_CFG_FILE} ${arg_WORKING_DIRECTORY}/.clang-format COPYONLY)
+
+    if(_generate_target)
+        # ClangFormat does not support --dry-run until version 10 which isn't on many machines.
+        # For now, use run-clang-format for dry running purposes.
+        if(${arg_MODIFY_FILES})
+            add_custom_target(${arg_NAME}
+                    COMMAND  ${CLANGFORMAT_EXECUTABLE} ${arg_PREPEND_FLAGS}
+                        --style=file -i ${arg_SRC_FILES} ${arg_APPEND_FLAGS}
+                    WORKING_DIRECTORY ${_wd} 
+                    COMMENT "${arg_COMMENT}Running ClangFormat source code formatting checks.")
+        else()
+            set(_run_clangformat "${BLT_ROOT_DIR}/cmake/run-clang-format.py" --clang-format-executable ${CLANGFORMAT_EXECUTABLE})
+            add_custom_target(${arg_NAME}
+                    COMMAND ${_run_clangformat} -j1 ${arg_SRC_FILES}
```

Comment:
Is the ``j`` flag necessary?

In other cases, with ``make``-based generators, I'm pretty sure I was able to run
```
make -j16 format
```
and it would run the formatting in parallel.
(I could be wrong though).

Reply:
It defaults the CPU +1 for the j.. I found that to be kinda scary.